### PR TITLE
feat(tile): added tile component

### PR DESCRIPTION
**Pull request by marijohannessen**
Added Tile component.

Consists of
- Tile
- ClickableTile
- SelectableTile
- ExpandableTile
   - TileAboveTheFoldContent
   - BelowTheFoldContent

### CODE REVIEW

**Comment by tw15egan** [comment]
👍 ✅

**Comment by chrisdhanaraj** [comment]
Quick skim - I'll do a deep dive, but two quick comments

- state needs to be able to be passed in via props as well
- any event handlers we put in needs to accept event handlers passed in via prop as well

File: components/Tile.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+  static defaultProps = {
+    tabIndex: 0,
+    expanded: false,
+    tileMaxHeight: '0',
+    handleClick: () => {},
+  }
+
+  componentDidMount = () => {
+    this.aboveTheFold = ReactDOM.findDOMNode(this.refs[0]); // eslint-disable-line
```
Comment (chrisdhanaraj) [inline components/Tile.js]: use a `ref` here instead of findDOMNode 

you'll also need to wrap any `ref` usage in an `if`

**Comment by chrisdhanaraj** [comment]
Bluemix [toolchain](https://console.bluemix.net/devops/toolchains/6b8b7163-1ec8-4616-82fa-aa467a17af03 "storybook-toolchain-v2"): [Delivery Pipeline](https://console.bluemix.net/devops/pipelines/b84aaac8-8484-4a1e-902a-ebb526eb7998 "storybook-toolchain-v2") deployed [carbon-components-react](https://github.com/carbon-design-system/carbon-components-react/releases/tag/deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20171011-005532 "deploy-ibm-yp-us-south-chrisdhanaraj_org-Bluemix-Design-System-20171011-005532") to [Bluemix Design System](https://console.bluemix.net/apps/bc75a101-6472-481c-b48d-3eed1e9703f3?env_id=ibm:yp:us-south "ibm:yp:us-south:chrisdhanaraj_org:Bluemix Design System"), including commit d2d64a2c005e00ef924db36462f90c04d57b00dd

**Pull request merged**